src/qengine/state/opencl.cpp

```diff
@@ -10,47 +10,59 @@
 // See LICENSE.md in the project root or https://www.gnu.org/licenses/gpl-3.0.en.html
 // for details.
 
+#include <memory>
+
 #include "oclengine.hpp"
 #include "qengine_opencl.hpp"
 
 namespace Qrack {
 
 #define CMPLX_NORM_LEN 5
 
-void QEngineOCL::InitOCL()
+void QEngineOCL::SetDevice(const int& dID)
 {
-    clObj = OCLEngine::Instance();
+    if (dID >= 0) {
+        deviceID = dID % (clObj->GetDeviceCount());
```

Comment:
Way better to abort here than to try to "make it work" by choosing some device essentially at random via modulus.

Reply:
If QEngineOCLMulti knows how many devices there are and how it wants to distribute load to them, yes. It doesn't, yet.